### PR TITLE
[Outreachy]Typeahead Search

### LINKED ISSUE #8838: Implement Typeahead on The Dashboard Topics Search
This issue is part of #8513

It will fix:

- [ ] Implements a typeahead search on the 'Follow more topics section'

### PR DESCRIPTION
**Pull request by RuthNjeri**
Fixes #8838

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

Screenshot 2021-01-22 at 23 00 55

### CODE REVIEW

**Comment by jywarren** [comment]
Ooooohhhhhh!!!!!! Ok, reviewing this now!

**Review by jywarren — APPROVED** [review]
This looks great. Unfortunately https://stable.publiclab.org is offline right now but this might bear some performance testing with a full database. That said, if it's all the same code as at https://publiclab.org/tags, then we should be OK!!!

File: app/views/dashboard_v2/_sidebar.html.erb [hunk 1]
```diff
@@ -47,7 +48,9 @@
               <% end %>
           </div>
         </div>
-        <%# TODO Search topics %>
+        <%# Search topics %>
```
Comment (jywarren) [inline app/views/dashboard_v2/_sidebar.html.erb:51]:
```suggestion
```

File: app/views/dashboard_v2/_sidebar.html.erb [hunk 2]
```diff
@@ -47,7 +48,9 @@
...
+        <br/><br/>
```
Comment (jywarren) [inline app/views/dashboard_v2/_sidebar.html.erb:52]:
```suggestion
        <h3>Search topics</h3>
```

What if we did this, and also converted other instances of `<%# ... %>` into regular HTML comments like `<!-- ... -->`? I think that could be nice to leave the rest of the page just normal HTML and not insert Ruby just to make a comment. Not a big deal, but stylistically nice!

**Review by jywarren — CHANGES_REQUESTED** [review]

**Comment by jywarren** [comment]
These changes were so small i just added a commit. Then we can merge!

**Comment by RuthNjeri** [comment]
Thanks @jywarren! Using  the HTML style for comments does make sense...I'll do that in the future.

**Pull request merged**